public/components/notebooks/components/paragraph_components/paragraphs.tsx

```diff
@@ -217,23 +226,25 @@ export const Paragraphs = forwardRef((props: ParagraphProps, ref) => {
     return newVizObject;
   };
 
-  const onRunPara = () => {
+  const onRunPara = (evt) => {
+    evt.stopPropagation();
```

Comment:
onRunPara needs to prevent the Paragraph-level `onClick` that selects paragraph - selecting paragraph sets state and causes a render - which stops this click-handler unless stopPropagation().